### PR TITLE
[build] allow to specify ninja targets to build for LLVM

### PR DESCRIPTION
**Pull request by edymtt**
This is meant to support scenarios in which we need to build as little as
LLVM as possible for performance reasons (e.g. when enabling LTO).

While LLVM CMake build system offers options in this sense,
in our investigation they turned out not to be suitable,
since either they are not granular enough (`LLVM_INCLUDE/BUILD` flags)
or they require active opt out for any new tool added
(`*_BUILD_*_TOOL` flags)

When using this mechanism, there is the possibility to specify different
targets to use for cross-compile hosts.

Supports rdar://32019390

### CODE REVIEW

**Comment by edymtt** [comment]
@swift-ci please smoke test

File: utils/build-script-impl [hunk 1]
```diff
@@ -1636,23 +1638,31 @@ for host in "${ALL_HOSTS[@]}"; do
...
-                          FileCheck
-                          not
-                          llvm-nm
-                      )
+                if [ "${LLVM_NINJA_TARGETS}" != "" ] ; then
+                    if [[ $(is_cross_tools_host ${host}) && "${LLVM_NINJA_TARGETS_FOR_CROSS_COMPILE_HOSTS}" != "" ]] ; then
+                        build_targets=("${LLVM_NINJA_TARGETS_FOR_CROSS_COMPILE_HOSTS[@]}")
+                    else
+                        build_targets=("${LLVM_NINJA_TARGETS[@]}")
+                    fi
```
Comment (compnerd) [inline utils/build-script-impl:1646]: I think that adding the additional check would make this clearer:

```
if [[ -n "${LLVM_NINJA_TARGETS_FOR_CROSS_COMPILE_HOSTS} && $(is_cross_tools_host ${host}) ]] ; then
  build_targest=( "${LLVM_NINJA_TARGETS_FOR_CROSS_COMPILE_HOSTS[@]}" )
elif [[ -n "${LLVM_NINJA_TARGETS}" ]] ; then
  build_targest=( "${LLVM_NINJA_TARGETS[@]}" )
else
  ...
fi
```

Comment (edymtt) [reply utils/build-script-impl]: Makes sense, that additional check does not bring any value.

Amended in e98ce3db4eb

**Comment by edymtt** [comment]
@swift-ci please smoke test

**Review by compnerd — APPROVED** [review]

**Comment by edymtt** [comment]
@swift-ci please smoke test macOS

**Comment by edymtt** [comment]
@swift-ci please smoke test macOS

**Comment by edymtt** [comment]
@swift-ci please smoke test

**Comment by edymtt** [comment]
Modified my PR so that flags that tell not to build all of LLVM have precedence over the newly introduced flags.

**Pull request merged**